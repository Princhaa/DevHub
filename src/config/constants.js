import { Dimensions, StyleSheet } from 'react-native'

const { width, height } = Dimensions.get('window')

const styles = StyleSheet.create({
  tabBarIcon: {
    width: 25,
    height: 25
  },

  shadow: {
    shadowColor: 'rgba(184, 184, 184, 0.5)',
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowRadius: 4,
    shadowOpacity: 1,
    elevation: 5,
  }

})

export default {
  DEVICE_WIDTH: width,
  DEVICE_HEIGHT: height,
  PRIMARY_COLOR: '#39A9DB',
  PRIMARY_COLOR_LIGHT: '#40BCD8',
  ACCENT_COLOR: '#008BA3',
  SECONDARY_COLOR: '#F39237',
  TERTIARY_COLOR: '#D63230',
  GREY: '#B0B0B0',
  SHADOW_COLOR: 'rgba(184, 184, 184, 0.5)',
  TAB_BAR_ICON: styles.tabBarIcon,
  SHADOW_STYLE: styles.shadow,
}
