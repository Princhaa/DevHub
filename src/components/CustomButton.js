/* @flow weak */

import React from 'react'
import { TouchableOpacity, Text, StyleSheet } from 'react-native'

import constants from '../config/constants'

const CustomButton = ({ children, color, ...otherProps }) => (
  <TouchableOpacity
    style={[styles.container, { backgroundColor: color }]}
    {...otherProps}
  >
    {children}
  </TouchableOpacity>
)

export default CustomButton

const styles = StyleSheet.create({
  container: {
    width: constants.DEVICE_WIDTH * 0.9,
    height: 40,
    borderRadius: 5,
    justifyContent: 'center',
    alignItems: 'center',
    marginVertical: 5
  }
})
