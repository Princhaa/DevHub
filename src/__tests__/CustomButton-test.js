import React from 'react'
import { Text } from 'react-native'
import ShallowRenderer from 'react-test-renderer/shallow'

import constants from '../config/constants'

import CustomButton from '../components/CustomButton'

it('Renders button correctly', () => {
  const renderer = new ShallowRenderer()
  renderer.render(
    <CustomButton color={constants.PRIMARY_COLOR}>
      <Text>Button</Text>
    </CustomButton>
  )
  const tree = renderer.getRenderOutput()
  expect(tree).toMatchSnapshot()
})
