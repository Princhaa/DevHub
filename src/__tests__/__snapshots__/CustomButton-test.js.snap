// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders button correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "height": 40,
        "justifyContent": "center",
        "marginVertical": 5,
        "width": 675,
      },
      Object {
        "backgroundColor": "#008BA3",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Button
  </Text>
</TouchableOpacity>
`;
